src/java.desktop/unix/classes/sun/print/CUPSPrinter.java

```diff
@@ -458,15 +472,34 @@ public static int getPort() {
         return cupsPort;
     }
 
+    /**
+     * Returns CUPS original (a fully-qualified hostname,
+     * a numeric IPv4 or IPv6 address, or a domain socket pathname)
+     * server name.
+     */
+    private static String getOriginalServer() {
+        return cupsOriginalServer;
+    }
+
+    static boolean useDomainSocketPathname() {
+        return isSandbox && PrintServiceLookupProvider.isMac()
+                && getOriginalServer().startsWith("/");
```

Comment:
AFAICT `getOriginalServer()` can return null here, since `cupsOriginalServer = getCupsServer(true)` and `getCupsServer()` can return null

Reply:
The fix is updated and cupsServer is checked to null when domainSocketPathname is initialized.